### PR TITLE
tests: Introduce retry mechanism for helm install

### PR DESCRIPTION
**Pull request by BbolroC**
Kata-deploy often fails due to a transiently unreachable k8s cluster for the qemu-coco-dev test on s390x.
```
Final kata-deploy manifests used in the test
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /home/ansible/.kube/config
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /home/ansible/.kube/config
install.go:222: [debug] Original chart version: ""
install.go:239: [debug] CHART PATH: /home/ansible/gha-runner/_layout/_work/kata-containers/kata-containers/tools/packaging/kata-deploy/helm-chart/kata-deploy

Error: INSTALLATION FAILED: Kubernetes cluster unreachable: Get "https://127.0.0.1:6443/version": dial tcp 127.0.0.1:6443: connect: connection refused
helm.go:84: [debug] Get "https://127.0.0.1:6443/version": dial tcp 127.0.0.1:6443: connect: connection refused
Kubernetes cluster unreachable
```
(e.g. https://github.com/kata-containers/kata-containers/actions/runs/10831142906/job/30058527098?pr=10009) 

This PR introduces a retry mechanism to mitigate these failures by retrying the command two more times with a 10-second interval as a workaround.

Signed-off-by: Hyounggyu Choi <Hyounggyu.Choi@ibm.com>

### CODE REVIEW

**Review by stevenhorsman — APPROVED** [review]
Looks okay to me. Thanks!

File: tests/integration/kubernetes/gha-run.sh [hunk 1]
```diff
@@ -207,7 +207,25 @@ function deploy_kata() {
...
 	echo "::endgroup::"
 
-	helm install kata-deploy "${helm_chart_dir}" --values "${values_yaml}" --namespace kube-system --debug
+	local max_tries=3
+	local interval=10
+	local i=0
+	# Retry loop for helm install to prevent transient failures due to instantly unreachable cluster
+	set +e # Disable immediate exit on failure
+	while true; do
+		helm install kata-deploy "${helm_chart_dir}" --values "${values_yaml}" --namespace kube-system --debug
```
Comment (fidencio) [inline tests/integration/kubernetes/gha-run.sh:216]: i am worried that helm will fail if rbac gets deployed, for instance ... in this case we better just wait more than retrigger the install.

Comment (BbolroC) [reply tests/integration/kubernetes/gha-run.sh:216]: What about using `helm upgrade --install ` instead of `helm install` then?

Comment (fidencio) [reply tests/integration/kubernetes/gha-run.sh]: sounds better!

Comment (BbolroC) [reply tests/integration/kubernetes/gha-run.sh]: Thanks for the comment. I missed thinking those remaining resources in the first try.

**Comment by BbolroC** [comment]
The CI failure for `qemu-coco-dev` is a false-negative result in the `Run tests` step, which has nothing to do with this PR.

**Review by fidencio — APPROVED** [review]
lgtm, thanks @BbolroC !

**Pull request merged**